### ISSUE TITLE
Webhooks retries policy

### ISSUE THREAD

**Issue by albertisfu** [freelawproject/courtlistener#2207]
As we commented on #1636 it's necessary to design a retry policy for webhooks.

Here are two options we could use:

**Use celery retries as our exponential backoff policy and count errors.**

Here we could take advantage of the celery retry system by adding an exponential backoff policy for the task, so in case of a failure, celery will retry automatically following an exponential backoff until the task succeeds or the max num of retries is reached.

The problem here is that since the Webhook event is not created until the task succeeds, we wouldn't be able to inform users about failed events until it succeeds or definitely fails. But we might take a different approach to create the WebhookEvent before doing the request and using the celery retries counter (as described [here](https://stackoverflow.com/a/27951681/64911)) to inform users of the number of current retries.

**I'll do some tests to assess the option that seems easier to implement.**

There is also a different approach:
**Don’t retry celery tasks and use an external exponential backoff retry method:**

- Don’t retry the `send_docket_to_webhook` celery task if it fails, so the WebhookEvent is created and it’ll be shown on the user's webhook panel as failed.
- We can use an external exponential backoff retry similar that the one we use to retry emails so that when a Webhook event fails we schedule a new task that will retry the failed webhook after e.g: 2 hours, if it fails again it would be retried after 4, 8, 16, 32 hours, etc.
- So we could inform users about the WebhookEvent status, when it’s going to be retried again or if no more retries will be attempted.

**Comment by mlissner** [comment]
I'm happy with either of these approaches. Your second one seems simpler to debug, I think? 

For the actual policy, I played with some numbers, aiming to terminate after about two days. Here's what I came up with:

Retry Count | New Delay | Elapsed
-- | -- | --
1 | 0:03 | 0:03
2 | 0:09 | 0:12
3 | 0:27 | 0:39
4 | 1:21 | 2:00
5 | 4:03 | 6:03
6 | 12:09 | 18:12
7 | 36:27 | 54:39

That's a multiplier of three with a starting backoff of three. It seems about right when I play it out in my head: 

> If a server is down, try again in a few minutes. Still down? Try another ten or so. Still? Try in half an hour. Still down? Try an hour or two. Still? Four hours. Still? Try about half a day. Still? Try a day or two. Still? Give up! 

And we can say something like, "It'll try with an exponential backoff and give up after about two days."
